railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -63,10 +64,12 @@ def gemspec
     end
 
     def gitignore
-      template "gitignore", ".gitignore"
+      template "gitignore", ".gitignore" unless inside_application?
```

Comment:
```
/.bundle/
/doc/
/log/*.log
/pkg/
/tmp/
```
I think the only useful line in here would be `/doc/` and that would only make sense if someone is putting docs in there. If you think its worth generating for that I can leave it but to me it doesn't seem like a file every plugin-in-application should have.

Reply:
I think it can make sense to add docs to a nested plugin.  The nested plugin might be the first step of extracting the code as a separate gem.  Or there might be separate developers working on the plugin and the app, and the documentation is for the app team.

Also, if the plugin is eventually extracted as a separate gem / repo, a `.gitignore` would be one less file for the user to have to create.

And then from our side of things, if we ever add more to the plugin `gitignore.tt`, we won't have to worry about whether it's covered by the app `gitignore.tt`.